From 79ac43bf178e2cadfcb9e624ba2f1d3a1b948c38 Mon Sep 17 00:00:00 2001
From: fanglifei <fanglifei@eswincomputing.com>
Date: Thu, 23 May 2024 13:14:03 +0800
Subject: [PATCH 015/222] fix(vo): add dma-noncoherent for video ouput device

Changelogs:
1. Fix display corruption by adding dma-noncoherent dts proporty
---
 arch/riscv/boot/dts/eswin/eswin-win2030-die0-soc.dtsi | 1 +
 arch/riscv/boot/dts/eswin/eswin-win2030-die1-soc.dtsi | 1 +
 2 files changed, 2 insertions(+)

diff --git a/arch/riscv/boot/dts/eswin/eswin-win2030-die0-soc.dtsi b/arch/riscv/boot/dts/eswin/eswin-win2030-die0-soc.dtsi
index 453f15acbd15..1e813abf0819 100644
--- a/arch/riscv/boot/dts/eswin/eswin-win2030-die0-soc.dtsi
+++ b/arch/riscv/boot/dts/eswin/eswin-win2030-die0-soc.dtsi
@@ -1739,6 +1739,7 @@ d0_graphcard2: graphcard2 {
 		video_output: display-subsystem {
 			compatible = "eswin,display-subsystem";
 			ports = <&dc_out>;
+			dma-noncoherent;
 		};
 
 		dvb_widgets: dvb-subsystem {
diff --git a/arch/riscv/boot/dts/eswin/eswin-win2030-die1-soc.dtsi b/arch/riscv/boot/dts/eswin/eswin-win2030-die1-soc.dtsi
index b1b077ceea0e..5a55de6c7b2a 100644
--- a/arch/riscv/boot/dts/eswin/eswin-win2030-die1-soc.dtsi
+++ b/arch/riscv/boot/dts/eswin/eswin-win2030-die1-soc.dtsi
@@ -1647,6 +1647,7 @@ d1_spi_demo: spi-demo@0 {
         d1_video_output: display-subsystem {
             compatible = "eswin,display-subsystem";
             ports = <&d1_dc_out>;
+            dma-noncoherent;
         };
 
         d1_dc: display_control@702c0000 {
-- 
2.47.0

